### PR TITLE
[fixes #425] rotate old keypackages

### LINKED ISSUE #425: Check on launch for outdated key packages without encoding tags and delete them from relays
As mentioned by @erskingardner in https://github.com/marmot-protocol/mdk/pull/98

### PR DESCRIPTION
**Pull request by dannym-arx**
## Summary by CodeRabbit

* **New Features**
  * Key package maintenance now detects packages missing encoding metadata, treats them as outdated ahead of expired packages, and will delete or replace them as needed.
  * Maintenance logging now clearly distinguishes rotations for expired vs. outdated packages.

* **Tests**
  * Added tests covering detection, deletion, and rotation/replacement of outdated packages and expiry boundary cases.
  * Added comprehensive unit tests for retry/backoff behavior.

### CODE REVIEW

File: src/whitenoise/scheduled_tasks/tasks/key_package_maintenance.rs [hunk 1]
```diff
@@ -421,4 +494,139 @@ mod tests {
...
+
+        // The new package should have the encoding tag
+        assert!(
+            has_encoding_tag(&after[0]),
+            "New key package should have encoding tag"
+        );
+    }
+
+    #[test]
+    fn test_find_outdated_packages_filters_correctly() {
```
Comment (jgmontoya) [inline src/whitenoise/scheduled_tasks/tasks/key_package_maintenance.rs]: This test feels out of place here, and redundant with what's already tested in `src/whitenoise/key_packages.rs`, no?

Comment (dannym-arx) [reply src/whitenoise/scheduled_tasks/tasks/key_package_maintenance.rs]: honestly, yes, but just tried to get the test coverage up lol

Comment (jgmontoya) [reply src/whitenoise/scheduled_tasks/tasks/key_package_maintenance.rs]: well I don't think this one is helping hahaha

Comment (dannym-arx) [reply src/whitenoise/scheduled_tasks/tasks/key_package_maintenance.rs]: Removed it - you're right, it's fully covered by the tests in `key_packages.rs`.

File: src/whitenoise/scheduled_tasks/tasks/key_package_maintenance.rs [hunk 2]
```diff
@@ -131,10 +144,48 @@ async fn maintain_key_packages(whitenoise: &Whitenoise, account: &Account) -> Ma
...
+            // package rotation (30 days), or will be cleaned up once a valid package exists.
+            tracing::info!(
+                target: "whitenoise::scheduler::key_package_maintenance",
+                "Account {} has {} outdated key package(s) but no valid packages, publishing new one",
+                account.pubkey.to_hex(),
+                outdated_packages.len()
+            );
+            return publish_new_key_package(whitenoise, account).await;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (dannym-arx) [reply src/whitenoise/scheduled_tasks/tasks/key_package_maintenance.rs:195]: The expired package path (Case 3) already does apply the filter_locally_parseable_key_packages ownership check, which is the right thing for parseable packages. This design decision is well-documented in the comments.

File: src/whitenoise/key_packages.rs [hunk 3]
```diff
@@ -6,7 +6,92 @@ use nostr_sdk::prelude::*;
...
+///
+/// * `event` - The key package event to check
+///
+/// # Returns
+///
+/// Returns `true` if the event has the required encoding tag, `false` otherwise.
+pub(crate) fn has_encoding_tag(event: &Event) -> bool {
+    event.tags.iter().any(|tag| {
+        tag.kind() == TagKind::Custom("encoding".into()) && tag.content() == Some("base64")
+    })
```
[the remark this answers is not in the document]
Comment (dannym-arx) [reply src/whitenoise/key_packages.rs:34]: Replacing .into() with .to_string() would create Cow::Owned(String::from("encoding")) which allocates memory unecessarily. 

TagKind::Custom("encoding".into()) converts to into Cow::Borrowed("encoding"). better imo
